matchzoo/data_pack/data_pack.py

```diff
@@ -294,6 +294,35 @@ def drop_label(self):
         """
         self._relation = self._relation.drop(columns='label')
 
+    @_optional_inplace
+    def drop_invalid(self):
+        """
+        Remove rows from the data pack where the content is invalid.
+
+        :param inplace: `True` to modify inplace, `False` to return a modified
+            copy. (default: `False`)
+
+        Example:
+            >>> import matchzoo as mz
+            >>> data_pack = mz.datasets.toy.load_data()
+            >>> data_pack.append_text_length(inplace=True, verbose=0)
+            >>> data_pack.drop_invalid(inplace=True)
+        """
+        if not ('length_left' in self._left and 'length_right' in self._right):
+            raise ValueError(f"`length_left` or `length_right` is missed. "
+                             f"Please call `append_text_length` in advance.")
+        invalid_left = self._left.loc[
+            self._left.length_left == 0].reset_index()
+        invalid_right = self._right.loc[
+            self._right.length_right == 0].reset_index()
+        self._left = self._left[~(self._left.index.isin(invalid_left.id_left))]
```

Comment:
You are doing two inversions here. Instead, create `valid_left` for `length_left != 0`, and use it directly to slice `self._left`.

Reply:
Good idea~